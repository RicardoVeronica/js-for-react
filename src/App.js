import { useState } from 'react';

const App = () => {
  const [value, setValue] = useState(0)

  return (
    <div>
      <h1>{ value }</h1>
      <button onClick={ () => setValue(value - 1) }>Sub</button>
      <button onClick={ () => setValue(value - value) }>Restart</button>
      <button onClick={ () => setValue(value + 1) }>Add</button>
    </div>
  )
}

export default App
