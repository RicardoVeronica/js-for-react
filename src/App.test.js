import App from './App'
import { createSerializer } from 'enzyme-to-json';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() })
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }))

describe('Test App component', () => {
  const wrapper = shallow(<App />)

  test('should find App component', () => {

    expect(wrapper).toMatchSnapshot()

  })

  test('should return counter value', () => {

    const element = wrapper.find('h1').text()

    expect(element).toEqual('0')

  })

  test('click to sub button', () => {

    wrapper.find('button').at(0).simulate('click')

    const value = wrapper.find('h1').text()

    expect(value).toEqual('-1')

  })

  test('click to add button', () => {

    wrapper.find('button').at(2).simulate('click')

    const value = wrapper.find('h1').text()

    expect(value).toEqual('0')

  })

  test('click to reset button', () => {

    wrapper.find('button').at(2).simulate('click')
    wrapper.find('button').at(2).simulate('click')

    wrapper.find('button').at(1).simulate('click')

    const value = wrapper.find('h1').text()

    expect(value).toEqual('0')

  })
})
