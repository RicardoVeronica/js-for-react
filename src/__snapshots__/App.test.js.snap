// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component should find App component 1`] = `
<div>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}
  >
    Sub
  </button>
  <button
    onClick={[Function]}
  >
    Restart
  </button>
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
